### ISSUE TITLE
Blob is not defined.

### ISSUE THREAD

**Issue by Detnator-Neo** [SheetJS/sheetjs#2366]
xlsx > jszip.js > line 1506
Exception has occurred: Reference Error: Blob is not defined.

I'll be honest, I'm relatively new to JS and I'm not even sure what a Blob is, but from what I can tell from Mozilla's JS documentation it's a webAPI and therefore probably isn't available in node, right?

Meanwhile I'm using nodejs on macOS and this error started showing up just in the last couple of days.  I've been using sheetjs in node for months without issue so far.

Can't figure out why it's trying to make a new Blob now, when as far as I'm aware it's never done this before.  In the course of running my app, this occurs when I require('xlsx').

MacBook  Pro M1
macOS 11.5.2
VS Code 1.59.1
node v16.4.1

I keep everything up to date generally. I occasionally give OS updates a week or two before installing those.  VS Code updated about a week ago.  macOS update from 11.4 to 11.5.2 yesterday --  perhaps something in that OS upgrade broke this.  Just a vague guess as to the problem here, but one way or another for some reason it wants to create a new Blob and it can't... for the first time I'm aware of.

Screen shot attached.

<img width="896" alt="Screen Shot 2021-08-25 at 2 12 15 AM" src="https://user-images.githubusercontent.com/70407340/130746300-833f78a6-01d8-4c62-82b4-c7d74d640329.png">

I can't be the only person to be having this issue surely?  Hope this is helpful, and hope you can fix it...?

**Comment by reviewher** [comment]
Share the part of the code that is calling `XLSX.read` or `XLSX.readFile` or `XLSX.write` or `XLSX.writeFile` as well as the installed version of the library (the "version" field in `/Volumes/1 DT/Dev/Code/nodejs/aa_modules/node_modules/xlsx/package.json`)

**Comment by Detnator-Neo** [comment]
Apologies... I was looking at my list of versions of everything and knew I'd missed something but couldn't figure out what it was!   

First saw this with 0.17.1.  Thought it might have been the update from 0.17.0 to that that caused this, so I went back to 0.17.0 to try it and the issue remained.

The code calling this is the require:

`const xlsx = require('xlsx');`

If it helps, here's the call stack also...

<img width="455" alt="Screen Shot 2021-08-25 at 9 42 31 AM" src="https://user-images.githubusercontent.com/70407340/130811972-b93c6f01-956a-4d8e-a2fa-85da2c7c49c9.png">

Of course please let me know if there's anything else you need. Thanks!

**Comment by Detnator-Neo** [comment]
One other piece of info that might be helpful.

If I simply type `./app.js` in my Terminal (no debugging, just run straight through), my app seems to run through to completion without error.  It's only when I'm running it in VSCode that I get this error.  

Therefore it's possible, I presume, that this is some issue in VSCode and/or some incompatibility between xlsx and VSCode, but I kinda feel like that's a stretch.

Apologies if this is all a bit vague.  I'm not quite sure what to think or conclude on this, but sharing what I can in case it's helpful.

**Comment by Detnator-Neo** [comment]
Ugh... I'm an idiot.  I'm so sorry...   Right after posting the above comment, something occurred to me.  

Yesterday I turned on the "Pause on Caught Exceptions" breakpoint.  This is pausing there because of that.  If I continue running, the `catch(e)` catches it and moves on.  

I just stepped through, the `catch` caught it, the next `try` for `var Builder` also exceptions then the next `catch (e)` catches that and it successfully executes `exports.blob = false`, continues on and my app runs through successfully, just like when run from the command line.

I turned off "Pause On Caught Exceptions" and everything works.

Thanks for your prompt reply above.  Sorry for the noise.  I'll close this now.  😊
